### ISSUE TITLE
A11Y: OrderSummary component missing aria-label or aria-labelledby

### ISSUE THREAD

**Issue by zeusorion** [carbon-design-system/carbon-components-react#513]
## Detailed description

Elements containing a 'region' role and section element must be labeled with an aria-label or aria-labelledby to facilitate page navigation by screen reader users.

> Is this issue related to a specific component?
OrderSummary, OrderSummaryTotal, OrderSummaryFooter

All of the components listed above product a `<section>` tag that must have an aria-label or aria-labelledby.

> What browser are you working in?
Chrome Version 63.0.3239.132 (Official Build) (64-bit) 
OS X (El Capitan)

> What version of the Carbon Design System are you using?
v3.14.0

## Steps to reproduce the issue

Instantiate any of the components, check the dom, notice that the section tags are missing the aria labels.

**Comment by zeusorion** [comment]
Violations noted by the DAP (Dynamic Assessment Plugin) Chrome Plugin tool
<img width="741" alt="screen shot 2018-01-19 at 3 04 28 pm" src="https://user-images.githubusercontent.com/2263351/35171583-f1e3ffa6-fd29-11e7-9f3b-986b56c9bed1.png">
